[REVIEW] Add CHANGELOG.md

harrism opened the pull request:
From now on every PR author should add a human-readable explanation of the PR at the appropriate location in this file. This will be a pre-condition for merging the PR. This PR does so.

kkraus14 commented:
Can we start the changelog with 0.3.0? Not sure what value providing the changelog for 0.2.x and 0.1.x adds.

harrism commented:
I thought the same thing. That's fine with me. I could use some help filling out 0.3.0.

kkraus14 commented:
Yes, cc @randerzander and I should be able to jump in and help with this.

harrism commented:
I went through the PRs that went into 0.3.0 and added them. I also added the changelog to 0.4.0. Any further additions to 0.4.0 (and the date) should be added in another PR.

I think this PR is ready for review.

kkraus14 commented:
Will let @mike-wendt merge pending his review.

dantegd commented on CHANGELOG.md:
```diff
@@ -0,0 +1,65 @@
+
+## Improvements
+ 
+ - PR #354 source code refactored for better organization. CMake build system overhaul.
```
Perhaps mention `beginning of Cython bindings transition` that was part of the refactor PR.

harrism replied:
Done.

harrism commented:
rerun tests

randerzander commented:
rerun tests
